Controllers films users

NatsCracker opened the pull request.

Commit:
FilmoRate v1.0
- Реализована начальная логика с использованием SprigBoot

Commit:
FilmoRate v1.0
- Реализована начальная логика с использованием SprigBoot

Commit:
FilmoRate v1.0
- Реализована начальная логика с использованием SprigBoot

Commit:
FilmoRate v1.0
- Реализована начальная логика с использованием SprigBoot

Commit:
FilmoRate v1.0
- Реализована начальная логика с использованием SprigBoot

Commit:
FilmoRate v1.0
- Реализована начальная логика с использованием SprigBoot

Commit:
FilmoRate v1.0
- Реализована начальная логика с использованием SprigBoot

Commit:
FilmoRate v1.0
- Реализована начальная логика с использованием SprigBoot

Commit:
FilmoRate v1.1
- Исправление ошибок

Commit:
FilmoRate v1.2
- Исправление ошибок

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,42 @@
+@RequestMapping("/films") // Маршрут для обработки запросов /films
 public class FilmController {
-}
+
```
Лучше хранить в виде мапы

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,42 @@
+
+    @GetMapping // Метод GET для получения списка фильмов
+    public List<Film> getAllFilms() {
+        return films;
```
Лучше возвращать копию списка

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,42 @@
+    public Film updateFilm(@RequestBody Film film) {
+        boolean removed = films.removeIf(f -> f.getId() == film.getId());
+        if (!removed) {
+            throw new RuntimeException("Film with id=" + film.getId() + " not found for update");
```
Лучше создать кастомное исключение

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,46 @@
+@Slf4j // Логирование
+@RestController // Контроллер для REST API
+@RequestMapping("/users") // Маршрут для обработки запросов по адресу /users
+public class UserController {
```
Здесь аналогичные комментарии как в FilmController

Commit:
FilmoRate v1.3
- Переделан ValidationExceptionHandler в GlobalExceptionHandler;
- Добавлено кастомное исключение;
- В FilmController и UserController хранение изменено на Map'у;

Commit:
FilmoRate v1.3
- Переделан ValidationExceptionHandler в GlobalExceptionHandler;
- Добавлено кастомное исключение;
- В FilmController и UserController хранение изменено на Map'у;

Commit:
FilmoRate v1.3
- Переделан ValidationExceptionHandler в GlobalExceptionHandler;
- Добавлено кастомное исключение;
- В FilmController и UserController хранение изменено на Map'у;

Commit:
FilmoRate v1.3
- Переделан ValidationExceptionHandler в GlobalExceptionHandler;
- Добавлено кастомное исключение;
- В FilmController и UserController хранение изменено на Map'у;

Commit:
FilmoRate v1.3
- Переделан ValidationExceptionHandler в GlobalExceptionHandler;
- Добавлено кастомное исключение;
- В FilmController и UserController хранение изменено на Map'у;

Commit:
FilmoRate v1.3
- Переделан ValidationExceptionHandler в GlobalExceptionHandler;
- Добавлено кастомное исключение;
- В FilmController и UserController хранение изменено на Map'у;

Commit:
Merge remote-tracking branch 'origin/controllers-films-users' into controllers-films-users

Commit:
FilmoRate v1.3
- Переделан ValidationExceptionHandler в GlobalExceptionHandler;
- Добавлено кастомное исключение;
- В FilmController и UserController хранение изменено на Map'у;

Commit:
Merge remote-tracking branch 'origin/controllers-films-users' into controllers-films-users

Commit:
FilmoRate v1.3
- Переделан ValidationExceptionHandler в GlobalExceptionHandler;
- Добавлено кастомное исключение;
- В FilmController и UserController хранение изменено на Map'у;

Commit:
Merge remote-tracking branch 'origin/controllers-films-users' into controllers-films-users
